### PR TITLE
refactor(app): extract trading limit validation

### PR DESCRIPTION
**Pull request by chapati23**
## The Problem
Issue #404 Phase (c) needs the swap form's trading-limit validation extracted out of `use-swap-form.tsx` into a tested pure module without changing the current behavior or message text.

## The Solution
- extract the trading-limit validation into `apps/app.mento.org/app/components/swap/swap-form/trading-limits.ts`
- keep `use-swap-form.tsx` wiring the same while delegating to the pure helper
- add focused tests for L0/L1/LG across in/out directions, implicit-limit wording, zero-limit behavior, and locale-driven timestamp formatting

Refs #404

Stacked on PR #468 (`refactor/404-max-sell-amount`). This PR delivers Phase (c) only.

## Validation
- `pnpm --filter app.mento.org test`
- `pnpm --filter app.mento.org exec tsc --noEmit`
- `trunk check --fix apps/app.mento.org/app/components/swap/swap-form/use-swap-form.tsx apps/app.mento.org/app/components/swap/swap-form/trading-limits.ts apps/app.mento.org/app/components/swap/swap-form/trading-limits.test.ts`

## Browser Verification
- started `app.mento.org` locally and loaded `http://localhost:3000/swap/celo`
- page rendered without a new runtime error from this change
- local console/network noise came from temporary verification env values and external fetches (`storage.valoraapp.com`, WalletConnect `projectId=local-dev`), not the extracted trading-limit logic

## Ship Checklist
- [x] ship skill used
- [x] local review run
- [x] validation run

---

> [!NOTE]
> **Low Risk**
> Refactor with no intended behavior or copy changes beyond an explicit null when `tokenToCheck` is unrecognized; validation is now unit-tested in isolation.
> 
> **Overview**
> Moves swap **trading-limit** checks out of `use-swap-form.tsx` into a new pure module `trading-limits.ts`, exporting `checkTradingLimitViolation` and `SwapTradingLimits`. The hook still drives form validation and error toasts but now calls the helper with an object argument instead of an inline `useCallback`.
> 
> The extracted logic keeps the same L0/L1/LG precedence, explicit vs implicit (token-in vs token-out) copy, zero limits as “not configured,” and locale-formatted amounts/dates. A small behavior tweak: if `tokenToCheck` matches neither input nor output symbol, the helper returns `null` instead of falling through silently.
> 
> Adds `trading-limits.test.ts` with Vitest coverage for each tier and direction, global limits, and zero-tier handling (mocking `@repo/web3`’s `parseAmountWithDefault`).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 05b3c04fa8690cb97e9e26cfa3ef8687db0d1e61. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: apps/app.mento.org/app/components/swap/swap-form/trading-limits.test.ts [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+
+vi.mock("@repo/web3", () => ({
+  parseAmountWithDefault: (
+    value: string | number | null | undefined,
+    defaultValue: string | number,
+  ) => {
+    const parsed = Number(value);
+    return Number.isFinite(parsed)
+      ? new ParsedAmount(parsed)
+      : new ParsedAmount(Number(defaultValue));
```
[the remark this answers is not in the document]
Comment (chapati23) [reply apps/app.mento.org/app/components/swap/swap-form/trading-limits.test.ts]: Fixed in 05b3c04f - moved the ParsedAmount test stub and parseAmountWithDefault helper into vi.hoisted(), so the mocked @repo/web3 factory no longer closes over a class initialized later in the file. Re-ran the narrow trading-limits Vitest file, full app.mento.org Vitest suite, app tsc --noEmit, Trunk on the changed file, git diff --check, and the pre-push hook; all passed.

**Comment by chapati23** [comment]
CI note: the failing `Visual Regression (app.mento.org)` check is an external Argos quota failure, not an app/test failure. Run `29080657266`, job `86322508711`: the app visual workflow completed 20 Playwright tests successfully, then Argos upload failed with HTTP 402: "You have reached the maximum screenshot capacity included in your Free Plan. Please upgrade your Plan." Build and Test, Trunk Check, Vercel previews, dependency checks, and Cursor are green; Claude/test workflow checks were still pending when this note was posted.
